gobblin-restli/gobblin-throttling-service/gobblin-throttling-service-client/src/main/java/gobblin/util/limiter/stressTest/MRStressTest.java

```diff
@@ -0,0 +1,363 @@
+
+  public static void main(String[] args) throws Exception {
+
+    CommandLine cli = StressTestUtils.parseCommandLine(OPTIONS, args);
+
+    Configuration configuration = new Configuration();
+    if (cli.hasOption(THROTTLING_SERVER_URI.getOpt())) {
+      configuration.setBoolean(USE_THROTTLING_SERVER, true);
+      String resourceLimited = cli.getOptionValue(RESOURCE_ID_OPT.getOpt(), "MRStressTest");
+      configuration .set(RESOURCE_ID, resourceLimited);
+      configuration .set(
+          BrokerConfigurationKeyGenerator.generateKey(new SharedRestClientFactory(),
+              new SharedRestClientKey(RestliLimiterFactory.RESTLI_SERVICE_NAME),
+              null, SharedRestClientFactory.SERVER_URI_KEY), cli.getOptionValue(THROTTLING_SERVER_URI.getOpt()));
+    }
+
+    if (cli.hasOption(LOCAL_QPS_OPT.getOpt())) {
+      configuration .set(LOCALLY_ENFORCED_QPS, cli.getOptionValue(LOCAL_QPS_OPT.getOpt()));
+    }
+
+    Job job = Job.getInstance(configuration, "ThrottlingStressTest");
+    job.getConfiguration().setBoolean("mapreduce.job.user.classpath.first", true);
+    job.getConfiguration().setBoolean("mapreduce.map.speculative", false);
+
+    job.getConfiguration().set(NUM_MAPPERS, cli.getOptionValue(NUM_MAPPERS_OPT.getOpt(), DEFAULT_MAPPERS));
+    StressTestUtils.populateConfigFromCli(job.getConfiguration(), cli);
+
+    job.setJarByClass(MRStressTest.class);
+    job.setMapperClass(StresserMapper.class);
+    job.setReducerClass(AggregatorReducer.class);
+    job.setInputFormatClass(MyInputFormat.class);
+
+    job.setOutputKeyClass(LongWritable.class);
+    job.setOutputValueClass(DoubleWritable.class);
+    FileOutputFormat.setOutputPath(job, new Path("/tmp/MRStressTest" + System.currentTimeMillis()));
+
+    System.exit(job.waitForCompletion(true) ? 0 : 1);
+  }
+
+  /**
+   * Instantiates a {@link Stressor} and runs it until it exits. It also sets up a {@link Recorder} that computes and
+   * records the rate at which the {@link AtomicLong} increases every 15 seconds.
+   */
+  public static class StresserMapper extends Mapper<Text, NullWritable, LongWritable, DoubleWritable> {
+    private SharedResourcesBroker<SimpleScopeType> broker;
+
+    @Override
+    protected void setup(Context context) throws IOException, InterruptedException {
+      Map<String, String> configMap = Maps.newHashMap();
+
+      SharedResourcesBrokerFactory.addBrokerKeys(configMap, context.getConfiguration());
+      this.broker = SharedResourcesBrokerFactory.createDefaultTopLevelBroker(ConfigFactory.parseMap(configMap),
+          SimpleScopeType.GLOBAL.defaultScopeInstance());
+
+      super.setup(context);
+    }
+
+    @Override
+    protected void map(Text key, NullWritable value, Context context) throws IOException, InterruptedException {
+      try {
+        Configuration configuration = context.getConfiguration();
+
+        Stressor stressor = context.getConfiguration().getClass(StressTestUtils.STRESSOR_CLASS,
```

Comment:
For now yes. I do see the point of having different stressors on each mapper, but I'd want to put that in a separate PR.